pennylane/qchem/openfermion_obs.py

```diff
@@ -837,6 +839,13 @@ def molecular_hamiltonian(
 
     |
 
+    .. warning::
+
+        If this function is called with a :code:`grouping_type` while :code:`enable_new_opmath()` is active, then
```

Comment:
It might be nice to link to the `enable_op_math` for users to have more context I guess.

Reply:
It's not super advertised yet. Maybe when we swap the default value to True we can link to it

Reply:
That is why I suggested we probably should not put the warning here. The user sees some unfamiliar names phrases that might be a bit confuting. But at the same time, not putting the warning here might cause confusion as well.